File: 1500-count-largest-group/count-largest-group.py
def find_digit(each):   
        sum_digit = 0
        while each:
            sum_digit += each % 10
            each//=10
        return sum_digit

class Solution(object):
    def countLargestGroup(self, n):
        groups = {} ##Creating dictionary 
        for each in range(1,n+1):        
            sum_dig = find_digit(each)
            if sum_dig not in groups:
                groups[sum_dig] =[]
            groups[sum_dig].append(each)       
        group_sizes = [len(group) for group in groups.values()]
        print(group_sizes)
        if group_sizes:
            max_size = max(group_sizes)
        else:
            max_size = 0

        count = group_sizes.count(max_size)
        #print(count)
        return count 
        

File: 1500-count-largest-group/count-largest-group.cpp
class Solution {

int digitsum(int num){
    int sum = 0;
    while(num > 0){
        sum += num % 10 ; 
        num/=10;
    }
    return sum;
}

public:
    int countLargestGroup(int n) {
        unordered_map<int, vector<int>> groups;
        for (int i = 1;i<=n; i++){
            int sum = digitsum(i);
            groups[sum].push_back(i) ;           
        }

        vector<int>sizes;
        for (const auto&pair:groups){
            sizes.push_back(pair.second.size());
        }

        if (sizes.empty()) return 0;
        int max_size = *max_element(sizes.begin(), sizes.end());

        int count = 0; 
        for (int size : sizes){
            if (size == max_size){
                count++;
            }
        }
        return count ;
    }
};